### DISCUSSION TITLE
Using the output of `run:` inside of `if:` condition?

### DISCUSSION THREAD

**Discussion by Cobertos** [community/community#25871] [Actions]
<p>Is there any way to get the output of a <code>run:</code> script into the <code>if:</code> condition? The feature I’m basically looking for is a step output from <code>run:</code> but that doesn’t look possible according to the documentation.</p>
<p>I’ve already gone the route of writing my own action to make a step with an output but it took a _lot_ of boilerplate because I needed to reauth with GitHub in the action, checkout the repository again, and do all my checks again in the action. I would have used <a><code>actions/github-script</code></a> but unfortunately the auth mechanism doesn’t currently meet my use case and it seemed if I had access to <code>run:</code> stuff inside <code>if:</code> this whole thing would’ve been much easier to accomplish</p>

**Comment by dawidd6** [comment, marked as answer]
<pre><code>- id: id
  run: echo ::set-output name=key::value
- if: steps.id.outputs.key == 'value'
</code></pre>
<p>This should work.</p>
